### PR TITLE
Migrate powershell docker scripts to docker@2

### PR DESCRIPTION
**Pull request by estebancams**
**Reason for Change**:

We are having intermittent failures when running docker powershell script in windows. Migrating to docker@2 is an attempt to fix this.
Even if this does not resolve the issue, still migrating to a native task is an improvement for pipelines

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: .pipelines/containers/container-template-windows.yaml [hunk 1]
```diff
@@ -11,19 +15,29 @@ steps:
...
-  retryCountOnTaskFailure: 3  
+  retryCountOnTaskFailure: 3
+  inputs:
+    command: 'build'
+    containerRegistry: $(ACR_SERVICE_CONNECTION)
+    repository: 'azure-${{ parameters.name }}'
+    tags: 'windows-${{ parameters.arch }}-${{ parameters.tag }}'
+    Dockerfile: '${{ parameters.name }}/windows.Dockerfile'
+    arguments: '
+    --build-arg VERSION=windows-${{ parameters.arch }}-${{ parameters.tag }}
```
Comment (camrynl) [inline .pipelines/containers/container-template-windows.yaml:28]: it looks like --build-arg is causing some errors in npm conformance test windows image builds

https://msazure.visualstudio.com/One/_build/results?buildId=63255597&view=logs&j=d229707a-47a3-50aa-1e61-a9a0b4c37ede&t=332ad56f-85e0-5dad-680f-4202a0f5b9d7&l=102

**Review by camrynl — CHANGES_REQUESTED** [review]

File: .pipelines/containers/container-template-windows.yaml (hunk 1 above)
Comment (estebancams) [reply .pipelines/containers/container-template-windows.yaml:28]: Thanks, fixed this issue

**Review by camrynl — APPROVED** [review]

**Review by camrynl — DISMISSED** [review]

**Review by rbtr — APPROVED** [review]

**Review by rbtr — APPROVED** [review]

**Pull request merged**